src/Microsoft.DotNet.Darc/Darc/Operations/VirtualMonoRepo/VmrDiffOperation.cs

```diff
@@ -0,0 +1,343 @@
+    }
+
+    private async Task VerifyInput(DiffRepo repo1, DiffRepo repo2)
+    {
+        if (!(repo1.IsVmr ^ repo2.IsVmr))
+        {
+            throw new DarcException("One of the repos must be a VMR, and the other one a product repo");
+        }
+
+        if (repo1.IsLocal)
+        {
+            var res = await processManager.ExecuteGit(repo1.Path, "status");
+            res.ThrowIfFailed($"{repo1.Path} is not a git repo");
+        }
+        if (repo2.IsLocal)
+        {
+            var res = await processManager.ExecuteGit(repo2.Path, "status");
+            res.ThrowIfFailed($"{repo2.Path} is not a git repo");
+        }
+    }
+        
+    private bool IsLocalRepoVmr(string repoPath) => fileSystem.FileExists(Path.Combine(repoPath, VmrInfo.SourceDirName, VmrInfo.SourceMappingsFileName));
+    private async Task<bool> IsRemoteRepoVmr(string uri, string branch)
+    {
+        var gitRepo = gitRepoFactory.CreateClient(uri);
+        try
+        {
+            await gitRepo.GetFileContentsAsync($"{VmrInfo.SourceDirName}/{VmrInfo.SourceMappingsFileName}", uri, branch);
+            return true;
+        }
+        catch(Exception)
+        {
+            return false;
+        }
+    }
+
+    private record DiffRepo(string Path, string Branch, bool IsLocal, bool IsVmr);
+
+    private async Task AddRemoteAndGenerateDiff(string repo1, string repo2, string repo2Branch, IReadOnlyCollection<string> filters)
+    {
+        string remoteName = Guid.NewGuid().ToString();
+
+        await processManager.ExecuteGit(repo1, [
+                "remote", "add", remoteName, repo2
+            ]);
+        var res = await processManager.ExecuteGit(repo1, [
+                "fetch", remoteName,
+                "--depth=1"
+            ]);
+        var sha1 = (await processManager.ExecuteGit(repo1, [
+                "rev-parse", "HEAD"
+            ])).StandardOutput.Trim();
+        var sha2 = (await processManager.ExecuteGit(repo1, [
+                "rev-parse", $"{remoteName}/{repo2Branch}"
+            ])).StandardOutput.Trim();
+
+        var patches = await patchHandler.CreatePatches(
+            options.OutputPath,
+            sha1,
+            sha2,
+            path: null,
+            filters,
+            relativePaths: false,
+            workingDir: new NativePath(repo1),
+            applicationPath: null,
+            CancellationToken.None);
+
+        if (patches.Count == 1)
+        {
+            using FileStream fs = new(patches[0].Path, FileMode.Open, FileAccess.Read);
+            using StreamReader sr = new(fs);
+            string? line;
+            while ((line = await sr.ReadLineAsync()) != null)
+            {
+                Console.WriteLine(line);
+            }
+        }
+        else
+        {
+            Console.WriteLine("Patch was too big so it had to be split into multiple files");
+            if (UxHelpers.PromptForYesNo("Do you want to apply the patches without seeing them?"))
```

Comment:
I think we will want to be able to use this command from a script so that people can write scripts to port changes.
This means there has to be some way to not do this interactive stuff.

Reply:
let's just output the `git apply` commands, and then people can use them however they want

Reply:
I'll leave the one Console.Writeline on top describing what's happening so they have something to check against

Reply:
BTW I don't think we should add an option that immediately applies these patches, you could just use the forwardflow/backward flow operations for that

Reply:
> let's just output the git apply commands, and then people can use them however they want

You can't really pipe this diff into anything if there's an arbitrary english sentence on top though

Reply:
> BTW I don't think we should add an option that immediately applies these patches, you could just use the forwardflow/backward flow operations for that

Those commands will give you a different result though. Forwardflow would just flow the delta, not overwrite the contents 1:1

Reply:
hmm in that case maybe a flag would be useful

Reply:
Either you need to provide some mode "output to files + list them / output patch contents / or this interactive way" or it needs to tell the user how it will behave (e.g. signal through exit code that there are more patches etc.)

Reply:
On Second thought, maybe we shouldn't do this, have the flag, since we can have both remote repos. Maybe we just go with what we have: output the diff, or output list of git apply commands if it's too big. Will add that to the Options description